Services/Mentor/OJS.Services.Mentor.Business/Implementations/MentorBusinessService.cs

```diff
@@ -0,0 +1,464 @@
+﻿namespace OJS.Services.Mentor.Business.Implementations;
+
+using System.Globalization;
+using System.Text;
+using DocumentFormat.OpenXml.Packaging;
+using DocumentFormat.OpenXml.Wordprocessing;
+using Microsoft.Build.Exceptions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using OJS.Common.Enumerations;
+using OJS.Common.Extensions;
+using OJS.Data.Models;
+using OJS.Data.Models.Mentor;
+using OJS.Servers.Infrastructure.Extensions;
+using OJS.Services.Common.Data;
+using OJS.Services.Infrastructure.Configurations;
+using OJS.Services.Infrastructure.Exceptions;
+using OJS.Services.Infrastructure.Extensions;
+using OJS.Services.Mentor.Business;
+using OJS.Services.Mentor.Models;
+using OpenAI;
+using OpenAI.Chat;
+using TiktokenSharp;
+using Table = DocumentFormat.OpenXml.Wordprocessing.Table;
+using static OJS.Common.GlobalConstants.Settings;
+
+public class MentorBusinessService : IMentorBusinessService
+{
+    private const string Docx = "docx";
+    private const string DocumentNotFoundOrEmpty = "Judge was unable to find the problem's description. Please copy and paste the problem description directly into the chat for the mentor to assist you.";
+
+    private readonly IDataService<UserMentor> userMentorData;
+    private readonly IDataService<MentorPromptTemplate> mentorPromptTemplateData;
+    private readonly OpenAIClient openAiClient;
+    private readonly IHttpClientFactory httpClientFactory;
+    private readonly IDataService<Setting> settingData;
+
+    public MentorBusinessService(
+        IDataService<UserMentor> userMentorData,
+        IDataService<MentorPromptTemplate> mentorPromptTemplateData,
+        IHttpClientFactory httpClientFactory,
+        IDataService<Setting> settingData,
+        IConfiguration configuration)
+    {
+        this.userMentorData = userMentorData;
+        this.mentorPromptTemplateData = mentorPromptTemplateData;
+        this.openAiClient = new OpenAIClient(configuration.GetSectionWithValidation<MentorConfig>().ApiKey);
```

Comment:
Shouldn't this client be initialized only once in service registration and reused?

Reply:
Done.